lib/element/appium-locator.js

```diff
@@ -20,11 +20,9 @@ class AppiumLocator extends LocateElement {
 
     const elementInstance = LocateElement.createElementInstance(element);
 
-    if (elementInstance.locateStrategy === 'id') {
-      return new By(elementInstance.locateStrategy, elementInstance.selector);
-    }
+    LocateStrategy.validate(elementInstance.locateStrategy);
```

Comment:
is this necessary? We already have this check at findElements level.

Reply:
Yeah, but those checks don't apply when the selector is passed as `{selector: 'some selector', locateStrategy: 'anything'}`. Earlier, this validation was done by `By[AVAILABLE_LOCATORS[elementInstance.locateStrategy]]`.